test/test_utils.py

```diff
@@ -122,6 +125,51 @@ def test_date_conversion():
     assert res == '2017-03-06'
     assert date_to_string(None) is None
 
+def test_get_query_param():
+    # Relative URL
+    next_url = '/api/v1/offerings?start=foo&limit=10'
+    next = get_query_param(next_url, 'start')
+    assert next == 'foo'
+    # Absolute URL
+    next_url = 'https://acme.com/api/v1/offerings?start=bar&limit=10'
+    next = get_query_param(next_url, 'start')
+    assert next == 'bar'
+    # Missing param
+    next_url = 'https://acme.com/api/v1/offerings?start=bar&limit=10'
+    next = get_query_param(next_url, 'token')
+    assert next is None
+    # No URL
+    next = get_query_param(None, 'start')
+    assert next is None
+    # Empty URL
+    next = get_query_param('', 'start')
+    assert next is None
+    # No query string
+    next_url = '/api/v1/offerings'
+    next = get_query_param(next_url, 'start')
+    assert next is None
+    # Bad query string
+    next_url = '/api/v1/offerings?start%XXfoo'
+    with pytest.raises(ValueError):
+        next = get_query_param(next_url, 'start')
+    # Duplicate param
+    next_url = '/api/v1/offerings?start=foo&start=bar&limit=10'
+    next = get_query_param(next_url, 'start')
+    assert next == 'foo'
+
+@pytest.mark.skipif(sys.version_info >= (3, 8), reason="Before Python 3.8, \
+     characters that affect netloc parsing under NFKC normalization were ignored.")
+def test_get_query_param_bad_url_legacy():
```

Comment:
I'm having a bit of trouble running this in my local environment.   For python 3.6 (3.6.13 to be precise), this test fails because the ValueError is being thrown similar to what is expected in the python 3.8+ case below.
Would it be better to just catch and absorb any ValueError that might be thrown inside the get_query_param method and just return None?